src/extension/kernel/kernel.ts

```diff
@@ -26,8 +30,98 @@ export class Kernel implements NotebookKernel {
     public readonly isPreferred: boolean = true;
     public readonly preloads: Uri[] = [];
     public readonly supportedLanguages?: string[] = ['kusto'];
-    constructor(public readonly document: NotebookDocument) {}
+    private static interactiveKernel?: Kernel;
+    public static get InteractiveKernel() {
+        return Kernel.interactiveKernel;
+    }
+    constructor(public readonly document: NotebookDocument) {
+        if (isKustoInteractive(document)) {
+            Kernel.interactiveKernel = this;
+        }
+    }
+    public async executeInteractiveSelection(textEditor: TextEditor): Promise<void> {
```

Comment:
Lot of copy & paste (similar code in `executeCell`).
Will re-factor and ensure its using common code, for now i've left this hacky approach in here.
Requires polishing.